src/System.Text.RegularExpressions/tests/RegexConstructorTests.cs

```diff
@@ -9,166 +9,30 @@
-            catch (ArgumentException)
-            {
-            }
-            catch (Exception e)
-            {
-                iCountErrors++;
-                Console.WriteLine("Err_2189jhss Expected Regex to throw ArgumentException and the following exception was thrown:\n {0}", e);
-            }
-
-            //[]RegEx with ECMA RegexOptions with all other valid options plus Singleline
-            strLoc = "Loc_9891asfes";
-            iCountTestcases++;
-            try
-            {
-                r = new Regex("foo", RegexOptions.ECMAScript | RegexOptions.IgnoreCase | RegexOptions.Multiline | RegexOptions.CultureInvariant | RegexOptions.Singleline);
-                iCountErrors++;
-                Console.WriteLine("Err_3156add Expected Regex to throw ArgumentException and nothing was thrown");
-            }
-            catch (ArgumentException)
-            {
-            }
-            catch (Exception e)
-            {
-                iCountErrors++;
-                Console.WriteLine("Err_456hjhj Expected Regex to throw ArgumentException and the following exception was thrown:\n {0}", e);
-            }
-
-            //[]RegEx with ECMA RegexOptions with all other valid options plus IgnorePatternWhitespace
-            strLoc = "Loc_23889asddf";
-            iCountTestcases++;
-            try
-            {
-                r = new Regex("foo", RegexOptions.ECMAScript | RegexOptions.IgnoreCase | RegexOptions.Multiline | RegexOptions.CultureInvariant | RegexOptions.IgnorePatternWhitespace);
-                iCountErrors++;
-                Console.WriteLine("Err_3568sdae Expected Regex to throw ArgumentException and nothing was thrown");
-            }
-            catch (ArgumentException)
-            {
-            }
-            catch (Exception e)
-            {
-                iCountErrors++;
-                Console.WriteLine("Err_4657dsacd Expected Regex to throw ArgumentException and the following exception was thrown:\n {0}", e);
-            }
-            ///////////////////////////////////////////////////////////////////
-            /////////////////////////// END TESTS /////////////////////////////
-        }
-        catch (Exception exc_general)
-        {
-            ++iCountErrors;
-            Console.WriteLine("Error Err_8888yyy!  strLoc==" + strLoc + ", exc_general==" + exc_general.ToString());
-        }
+        // Should not throw
+        new Regex("foo", RegexOptions.ECMAScript | RegexOptions.IgnoreCase | RegexOptions.Multiline | RegexOptions.CultureInvariant);
+    }
 
-        ////  Finish Diagnostics
-        Assert.Equal(0, iCountErrors);
+    [Fact]
+    public static void Ctor_Invalid()
+    {
+        // Pattern is null
+        Assert.Throws<ArgumentNullException>(() => new Regex(null));
+        Assert.Throws<ArgumentNullException>(() => new Regex(null, RegexOptions.None));
+        Assert.Throws<ArgumentNullException>(() => new Regex(null, RegexOptions.None, new TimeSpan()));
```

Comment:
My standard question: are there argument names we can validate in all of these argument exceptions? :)